arbor/include/arbor/recipe.hpp

```diff
@@ -54,8 +54,8 @@ struct cell_connection_base {
 
     cell_connection_base(L src, cell_local_label_type dst, float w, const U::quantity& d):
         source(std::move(src)), target(std::move(dst)), weight(w), delay(d.value_as(U::ms)) {
-        if (std::isnan(weight)) throw std::out_of_range("Connection weight must be finite.");
-        if (std::isnan(delay) || delay < 0)  throw std::out_of_range("Connection delay must be non-negative and infinite in units of [ms].");
+        if (std::isnan(weight)) throw std::domain_error("Connection weight must be finite.");
+        if (std::isnan(delay) || delay <= 0)  throw std::domain_error("Connection delay must be positive, finite, and given in units of [ms].");
```

Comment:
Inasmuch as we're presuming NaNs work as expected, we could just have the test `!(delay > 0)` instead of `std::isnan(delay) || delay <= 0`.

Do we really need to enforce that delay is finite? If so, then the test should include that.

Also, not being familiar (yet) with how LLNL units works, why do we need to specify that the quantity is in milliseconds? Can't we just convert as required or else assert in the type?

Reply:
Well, the two issues go hand in hand: `(42 * U.ms).value_as(U.mV) == NaN`. So, receiving a nan can mean either we got `nan * U.ms` or an erroneous unit. This is why the message allows for both.